src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -195,6 +241,11 @@ export class StudyViewPageStore {
         return this._cnaGeneFilter ? this._cnaGeneFilter.alterations : [];
     }
 
+    public getClinicalDataFiltersByUniqueKey(uniqueKey: string): string[] {
+        let filters = _.filter(this._clinicalDataEqualityFilterSet.values(), filter => _.isEqual(filter.clinicalDataType + '_' + filter.attributeId, uniqueKey));
```

Comment:
you could use _.find which will return stop iterating when it find the first item where predicate is true.  or undefined if none are found.